x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_store_data_client.ts

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger, ElasticsearchClient, SavedObjectsClientContract } from '@kbn/core/server';
+import type { EntityClient } from '@kbn/entityManager-plugin/server/lib/entity_client';
+
+import type {
+  InitEntityStoreRequestBody,
+  InitEntityStoreResponse,
+} from '../../../../common/api/entity_analytics/entity_store/engine/init.gen';
+import type {
+  EngineDescriptor,
+  EntityType,
+} from '../../../../common/api/entity_analytics/entity_store/common.gen';
+import { entityEngineDescriptorTypeName } from './saved_object';
+import { EngineDescriptorClient } from './saved_object/engine_descriptor';
+import { ensureEngineExists, getEntityDefinition } from './utils/utils';
+
+interface EntityStoreClientOpts {
+  logger: Logger;
+  esClient: ElasticsearchClient;
+  entityClient: EntityClient;
+  namespace: string;
+  soClient: SavedObjectsClientContract;
+}
+
+export class EntityStoreDataClient {
+  private engineClient: EngineDescriptorClient;
+  constructor(private readonly options: EntityStoreClientOpts) {
+    this.engineClient = new EngineDescriptorClient(options.soClient);
+  }
+
+  /**
+  * HOST:INIT
+    curl -H 'Content-Type: application/json' \
+      -X POST \
+      -H 'kbn-xsrf: true' \
+      -H 'elastic-api-version: 2023-10-31' \
+      http:///elastic:changeme@localhost:5601/api/entity_store/engines/host/init
+  */
+  public async init(
+    entityType: EntityType,
+    { indexPattern = '', filter = '' }: InitEntityStoreRequestBody
+  ): Promise<InitEntityStoreResponse> {
+    const definition = getEntityDefinition(entityType);
+
+    this.options.logger.debug(`Initializing entity store for ${entityType}`);
+
+    const savedObj = await this.engineClient.init(entityType, definition, filter);
+    await this.options.entityClient.createEntityDefinition({
+      definition: {
+        ...definition,
+        filter,
+        indexPatterns: [...definition.indexPatterns, ...indexPattern.split(',')],
+      },
+    });
+    const updatedObj = await this.engineClient.update(savedObj.id, 'started');
+
+    return { ...savedObj.attributes, ...updatedObj.attributes };
+  }
+
+  /**
+  * HOST:START
+    curl -H 'Content-Type: application/json' \
+      -X POST \
+      -H 'kbn-xsrf: true' \
+      -H 'elastic-api-version: 2023-10-31' \
+      http:///elastic:changeme@localhost:5601/api/entity_store/engines/host/start
+  */
+  public async start(entityType: EntityType) {
+    const definition = getEntityDefinition(entityType);
+
+    const savedObj = await this.engineClient.get(entityType).then(ensureEngineExists(entityType));
```

Comment:
I think it would be neater to have push this saved object client abstraction a _tiny_ bit further so we can "hide" the fact its a saved object at all and not have to deal with attributes and also make the code a bit more to the descriptive, e.g 

```
const engineDefinition = this.engineClient.getEngineDefinition(entityType).then(ensureEngineExists(entityType));

if(engineDefintion.status !== 'stopped')
```

Reply:
Done. 
I managed to hide the saved object isntance away so the Store client doesn't know what the udnerlying implementation is.

I left the `find` and `get` methods on the `Engine` client for more flexibility.